interface-impl-amf/src/main/java/org/mule/amf/impl/parser/factory/AMFParserWrapper.java

```diff
@@ -6,46 +6,69 @@
  */
 package org.mule.amf.impl.parser.factory;
 
-import amf.MessageStyle;
-import amf.ProfileName;
-import amf.client.AMF;
-import amf.client.environment.Environment;
-import amf.client.model.document.Document;
-import amf.client.parse.Parser;
-import amf.client.resolve.Resolver;
-import amf.client.validate.ValidationReport;
-import amf.plugins.document.webapi.resolution.pipelines.AmfResolutionPipeline;
+import amf.apicontract.client.platform.AMFBaseUnitClient;
+import amf.apicontract.client.platform.AMFConfiguration;
+import amf.apicontract.client.platform.APIConfiguration;
+import amf.core.client.common.transform.PipelineId;
+import amf.core.client.platform.AMFParseResult;
+import amf.core.client.platform.config.RenderOptions;
+import amf.core.client.platform.execution.ExecutionEnvironment;
+import amf.core.client.platform.model.document.BaseUnit;
+import amf.core.client.platform.model.document.Document;
+import amf.core.client.platform.validation.AMFValidationReport;
+import amf.core.client.platform.validation.AMFValidationResult;
 import org.mule.amf.impl.exceptions.ParserException;
+import org.mule.amf.impl.loader.ExchangeDependencyResourceLoader;
+import org.mule.amf.impl.loader.ProvidedResourceLoader;
+import org.mule.apikit.model.api.ApiReference;
+import org.yaml.builder.JsonOutputBuilder;
 
+import java.io.File;
 import java.net.URI;
 import java.net.URLDecoder;
+import java.util.List;
 import java.util.concurrent.CompletableFuture;
 
 public class AMFParserWrapper {
 
-  private Environment environment;
-  private Parser parser;
-  private Resolver resolver;
-  private ProfileName profileName;
-  private MessageStyle messageStyle;
-
-  public AMFParserWrapper(Environment environment, Parser parser, Resolver resolver, ProfileName profileName,
-                          MessageStyle messageStyle) {
-    this.environment = environment;
-    this.parser = parser;
-    this.resolver = resolver;
-    this.profileName = profileName;
-    this.messageStyle = messageStyle;
+
+  private final BaseUnit model;
+  private final AMFBaseUnitClient client;
+  private final List<AMFValidationResult> parsingIssues;
+  private final AMFConfiguration amfConfiguration;
+
+
+  public AMFParserWrapper(ApiReference apiRef, ExecutionEnvironment execEnv) {
+    AMFConfiguration amfConfiguration = APIConfiguration
+        .API()
+        .withExecutionEnvironment(execEnv);
+
+    if (apiRef.getResourceLoader().isPresent()) {
+      amfConfiguration = amfConfiguration.withResourceLoader(new ProvidedResourceLoader(apiRef.getResourceLoader().get()));
+    }
+    URI apiUri = apiRef.getPathAsUri();
+
+    if (apiUri.getScheme() != null && apiUri.getScheme().startsWith("file")) {
```

Comment:
why needed now ?

Reply:
This was also needed before to add the exchange resource loader but was moved.